### PR TITLE
GODRIVER-2323 Support int64 for 'n' field in insert, update, and delete ops.

### PR DESCRIPTION
**Pull request by matthewdale**
[GODRIVER-2323](https://jira.mongodb.org/browse/GODRIVER-2323)

Add support for int32 and int64 types in the `"n"` and `"nModified"` fields in various server CRUD responses.

### CODE REVIEW

**Review by benjirewis — COMMENTED** [review]
Should we add a similar type switch in [`count.go`](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/operation/count.go#L58) to not error on the receiving of an `int32` from the server?

**Comment by benjirewis** [comment]
There are also some lint failures for now-unnecessary conversions to `int64` in `collection.go`.

**Review by prestonvasquez — APPROVED** [review]

**Comment by matthewdale** [comment]
@benjirewis Added int32 support to `count` and fixed the linter failures.

File: x/mongo/driver/operation/count.go [hunk 1]
```diff
@@ -54,9 +55,12 @@ func buildCountResult(response bsoncore.Document) (CountResult, error) {
 	for _, element := range elements {
 		switch element.Key() {
 		case "n": // for count using original command
-			var ok bool
-			cr.N, ok = element.Value().AsInt64OK()
```
Comment (kevinAlbs) [inline x/mongo/driver/operation/count.go:58]: Please revert this change. `AsInt64OK` already supports reading `Int32`, `Int64`, and `Double`.

File: x/mongo/driver/operation/delete.go [hunk 2]
```diff
@@ -54,10 +55,13 @@ func buildDeleteResult(response bsoncore.Document) (DeleteResult, error) {
 	for _, element := range elements {
 		switch element.Key() {
 		case "n":
-			var ok bool
-			dr.N, ok = element.Value().AsInt32OK()
```
Comment (kevinAlbs) [inline x/mongo/driver/operation/delete.go:58]: Consider using `AsInt64OK()` to simplify here and in insert.go and update.go. `AsInt64OK()` also accepts Double. But I do not see any harm in accepting Double.

**Review by kevinAlbs — COMMENTED** [review]
Thank you for capturing the rationale in the ticket comments.

File: x/mongo/driver/operation/delete.go (hunk 2 above)
Comment (matthewdale) [reply x/mongo/driver/operation/delete.go:58]: Whoa, I didn't realize these are actually using a mixture of `AsInt__OK` and `Int__OK` and that `AsInt__OK` converts types already 🤯 . Great suggestion, will update all to use `AsInt64OK` everywhere!

File: x/mongo/driver/operation/count.go (hunk 1 above)
Comment (matthewdale) [reply x/mongo/driver/operation/count.go:58]: Great suggestion, will revert.

Comment (benjirewis) [reply x/mongo/driver/operation/count.go:58]: Very true! My bad.

**Review by benjirewis — APPROVED** [review]
LGTM after switching to `AsInt64Ok()`. Apologies for the misleading comment about `count.go`.

File: x/mongo/driver/operation/count.go (hunk 1 above)
Comment (matthewdale) [reply x/mongo/driver/operation/count.go]: Reverted.

**Review by kevinAlbs — APPROVED** [review]
LGTM!
